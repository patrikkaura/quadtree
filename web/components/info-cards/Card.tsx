import MuiCard from "@mui/material/Card";
import MuiCardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";

export type Props = {
  header: string;
  content: string | number;
  width: number;
  color: string;
};

export function Card({ header, content, width, color }: Props) {
  return (
    <MuiCard
      style={{
        width,
        borderTop: "5px solid",
        borderColor: color,
      }}
    >
      <MuiCardContent>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          {header}
        </Typography>
        <Typography sx={{ fontSize: 25 }}>{content}</Typography>
      </MuiCardContent>
    </MuiCard>
  );
}
