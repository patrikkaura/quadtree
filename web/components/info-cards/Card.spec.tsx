import { render } from "@testing-library/react";

import { Card, Props } from "./Card";

describe("Card", () => {
  let props: Props = {
    header: "header",
    content: "random-content",
  };

  it("should render", () => {
    const { container } = render(<Card {...props} />);
    expect(container).toMatchSnapshot();
  });
});
