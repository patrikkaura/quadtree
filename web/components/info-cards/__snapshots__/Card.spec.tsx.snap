// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="width: 200px; border-top: 5px solid; border-color: #000000;"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
      >
        header
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ll0ml8-MuiTypography-root"
      >
        random-content
      </p>
    </div>
  </div>
</div>
`;
